### ISSUE TITLE
Update SaveModelHistory output to json

### ISSUE THREAD

**Issue by johnypark** [johnypark/mytflib#2]
Currently it outputs csv, and only LR, loss, metric.

It would be beneficial to include additional info for training, such as 1. Model structure 2. Optimizer settings ( such as wd) 3. Loss functions 4. Other settings. 5. Output time taken for the jobs( for each step and the whole time spent).

More desirable format is json for that purpose.

**Comment by johnypark** [comment]
Update ```SaveModelHistory()```, now it saves the followings:
1. model.summary() to text file.
2. optimizer.get_config(), loss.get_config(), and other configuration settings to ```.json```
3. time taken for each epoch along with ```loss```, ```metric```, ```val_loss```, ```val_metric``` to csv file.

**Comment by johnypark** [comment]
Closing the issue for now, as these outputs work fine.
